includes/rest-api/class-sensei-rest-api-course-progress-controller.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * File containing Sensei_REST_API_Course_Progress_Controller class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Course Progress REST API Controller
+ *
+ * @since x.x.x
+ */
+class Sensei_REST_API_Course_Progress_Controller extends \WP_REST_Controller {
+
+	/**
+	 * Routes namespace.
+	 *
+	 * @var string
+	 */
+	protected $namespace;
+
+	/**
+	 * Routes prefix.
+	 *
+	 * @var string
+	 */
+	protected $rest_base = 'course-progress';
+
+	/**
+	 * Sensei_REST_API_Course_Structure_Controller constructor.
+	 *
+	 * @param string $namespace Routes namespace.
+	 */
+	public function __construct( $namespace ) {
+		$this->namespace = $namespace;
+	}
+
+	/**
+	 * Register the REST API endpoints for Course Structure.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			$this->rest_base . '/batch',
+			[
+				[
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => [ $this, 'batch_delete_items' ],
+					'permission_callback' => [ $this, 'batch_delete_items_permissions_check' ],
+					'args'                => $this->get_args_schema(),
+				],
+			]
+		);
+	}
+
+	/**
+	 * Reset progress for students on courses.
+	 *
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response
+	 */
+	public function batch_delete_items( WP_REST_Request $request ) {
+		$params      = $request->get_params();
+		$student_ids = $params['student_ids'];
+		$course_ids  = $params['course_ids'];
+		foreach ( $student_ids as $student_id ) {
+			$user = new WP_User( $student_id );
+			if ( $user->exists() ) {
+				foreach ( $course_ids as $course_id ) {
+					if ( Sensei_Utils::has_started_course( $course_id, $student_id ) ) {
+						Sensei_Utils::reset_course_for_user( $course_id, $student_id );
+					}
+				}
+			}
+		}
+
+		return new WP_REST_Response( null, WP_HTTP::OK );
```

Comment:
Should we update the response a bit so that the API caller knows for which students and which courses the operation was successful and which students were not found? Maybe like this

https://github.com/Automattic/sensei/blob/bc9f8fee08ad52dd4bb47ffb6e7bce2d15015f4e/includes/rest-api/class-sensei-rest-api-course-students-controller.php#L101-L114